### DISCUSSION TITLE
can you share an example of icdf with negative binomial and poisson distros?

### DISCUSSION THREAD

**Discussion by jeycervantes** [awslabs/gluonts#3230] [Q&A]
Hi,

I am looking into this library since it seems to support icdf for these distros but after installing  and using in databricks it says these methods aren't implemented?

I see they should be implemented here however: https://github.com/awslabs/gluonts/blob/3d05d465f97096583d87a6827948e0d0d3cb7e92/src/gluonts/torch/distributions/negative_binomial.py#L57 

 Sorry if I just missed it, is there some demo snippet of code around this use-case? Thank you!

Here is some sample code below  (that does not work)
from gluonts.mx import distribution
import mxnet as mx

# Example parameters for Negative Binomial distribution
n = mx.nd.array([10])
p = mx.nd.array([0.5])
# Create Negative Binomial distribution
neg_binom_dist = distribution.NegativeBinomial(mu=n, alpha=p)

# Generate samples
samples = neg_binom_dist.sample(num_samples=10000).asnumpy()

# Compute the inverse CDF (quantile function)
quantiles = neg_binom_dist.icdf(mx.nd.array([0.1, 0.5, 0.9])).asnumpy()

# Display the quantiles
display(quantiles)

<img width="445" alt="image" src="https://github.com/user-attachments/assets/86b16db4-ebd4-4efc-b393-bc35d8a37154">

Best,
J

**Comment by jeycervantes** [comment]
To clarify, am using the latest version of torch/gluonts

**Comment by lostella** [comment]
@jeycervantes you appear to be importing from the wrong path: use `gluonts.torch` instead of `gluonts.mx`, and contruct the distribution by passing in `torch.tensor` objects. Here is the negative binomial distribution, with the icdf method: https://github.com/awslabs/gluonts/blob/dev/src/gluonts/torch/distributions/negative_binomial.py#L57
